rad/src/lib.rs

```diff
@@ -42,22 +42,30 @@ pub struct RADRequestExecutionReport {
 pub async fn try_data_request(
     request: &RADRequest,
     settings: RadonScriptExecutionSettings,
+    inputs_injection: Option<&[&str]>,
 ) -> RADRequestExecutionReport {
     let context = &mut ReportContext::default();
 
-    let retrieve_responses_fut = request
-        .retrieve
-        .iter()
-        .map(|retrieve| run_retrieval_report(retrieve, settings));
-
-    let retrieval_reports: Vec<RadonReport<RadonTypes>> =
-        futures::future::join_all(retrieve_responses_fut)
```

Comment:
Now the big question: was the join_all needed to run the retrievals in parallel, or are they run in parallel by default?

Reply:
Well, it looks like this only affects the `tryDataRequest` method, the node uses a different method to execute the data requests.

Reply:
Yes, it looks like the `join_all` was needed to run the retrievals in parallel, can you open an issue and add a FIXME?